lib/jbuilder/jbuilder_template.rb

```diff
@@ -132,10 +132,12 @@ def array!(collection = EMPTY_ARRAY, *args, &block)
   def set!(name, object = BLANK, *args)
     options = args.first
 
-    if args.one? && _partial_options?(options)
+    if _partial_options?(options)
       _set_inline_partial name, object, options
+    elsif ::Kernel.block_given?
+      _set(name, object, args) { |x| yield x }
     else
-      super
+      _set(name, object, args)
     end
```

Comment:
This looks strange, but is the only way I found to avoid having a `&block` parameter.

An alternative was
```ruby
    elsif ::Kernel.block_given?
      super
    else
      _set(name, object, args)
    end
```

but this would incur an extra memory allocation for `*args` via the `super` call whenever a block was provided to `set!`.